### PR TITLE
add MeasureDao.selectProjectMeasuresByDeveloperForMetrics

### PR DESCRIPTION
**Pull request by sns-seb**
used by Dev Cockpit for DEVACT-163

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: sonar-db/src/test/java/org/sonar/db/measure/MeasureDaoTest.java [hunk 1]
```diff
@@ -415,10 +423,97 @@ public void selectByDeveloperForSnapshotAndMetrics_returns_empty_when_snapshotId
...
+    return moduleDto;
+  }
+
+  private ComponentDto insertProject(String uuid) {
+    ComponentDto projectDto = ComponentTesting.newProjectDto(uuid);
+    return insertComponent(projectDto);
+  }
+
+  private SnapshotDto insertSnapshot(ComponentDto componentDto, boolean last) {
+    SnapshotDto snapshotDto = new SnapshotDto().setComponentId(componentDto.getId()).setLast(last).setQualifier(componentDto.qualifier()).setScope(componentDto.scope());
```
Comment (julienlancelot) [inline sonar-db/src/test/java/org/sonar/db/measure/MeasureDaoTest.java]: You should use SnapshotTesting.newSnapshotForProject

File: sonar-db/src/test/java/org/sonar/db/measure/MeasureDaoTest.java [hunk 2]
```diff
@@ -415,10 +423,97 @@ public void selectByDeveloperForSnapshotAndMetrics_returns_empty_when_snapshotId
...
+    dbClient.snapshotDao().insert(dbSession, snapshotDto);
+    dbSession.commit();
+    return snapshotDto;
+  }
+
+  private MeasureDto insertMeasure(SnapshotDto snapshotDto, Long developerId, int metricId, double value) {
+    MeasureDto measureDto = new MeasureDto().setMetricId(metricId).setValue(value).setSnapshotId(snapshotDto.getId()).setDeveloperId(developerId);
```
Comment (julienlancelot) [inline sonar-db/src/test/java/org/sonar/db/measure/MeasureDaoTest.java]: You should use MeasureTesting.newMeasureDto

**Comment by julienlancelot** [comment]
LGTM

**Pull request closed without merging**